src/Compilers/VisualBasic/Portable/Lowering/LocalRewriter/LocalRewriter_FieldOrPropertyInitializer.vb

```diff
@@ -54,23 +54,50 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
-                If objectCreationExpression.InitializerOpt IsNot Nothing AndAlso
-                    objectCreationExpression.InitializerOpt.Kind = BoundKind.ObjectInitializerExpression Then
-                    objectInitializer = DirectCast(objectCreationExpression.InitializerOpt, BoundObjectInitializerExpression)
-                    createTemporary = objectInitializer.CreateTemporaryLocalForInitialization
-                End If
-            End If
-
             Dim instrument As Boolean = Me.Instrument(node)
 
             For symbolIndex = 0 To initializedSymbols.Length - 1
                 Dim symbol = initializedSymbols(symbolIndex)
                 Dim accessExpression As BoundExpression
 
+                Dim initialValueToRewrite As BoundExpression
+
+                If symbolIndex = 0 Then
+                    initialValueToRewrite = node.InitialValue
+                Else
+                    ' For all variables except the first one we rebind the initializer
+                    ' and throw away diagnostics, those are supposed to be reported in the first
+                    ' binding and don't need to be duplicated
+                    '
+                    ' Note that we have to rebind the initializers because current implementation of lambda 
+                    ' rewriter does not handle correctly blocks and locals which are reused in bound tree. 
+                    ' Actually it seems to heavily rely on an assumption that the bound tree is a tree, not 
+                    ' a DAG, with just minor deviations. Thus, the natural way of just rewriting bound 
+                    ' initializer stored in node.Initializer simply does not work because rewriting *may* 
+                    ' keep many blocks and locals unchanged and reuse them in all rewritten initializers, 
+                    ' in which case if we have a lambda inside the initializer lambda rewriter simply throws.
+
+                    If symbol.Kind = SymbolKind.Field Then
+                        ' Rebind and discard diagnostics
+                        initialValueToRewrite = node.BinderOpt.BindFieldInitializerExpression(syntax, DirectCast(symbol, FieldSymbol), BindingDiagnosticBag.Discarded)
+                    Else
+                        ' Rebind and discard diagnostics
+                        initialValueToRewrite = node.BinderOpt.BindPropertyInitializerExpression(syntax, DirectCast(symbol, PropertySymbol), BindingDiagnosticBag.Discarded)
+                    End If
+                End If
+
+                Dim objectInitializer As BoundObjectInitializerExpression = Nothing
+                Dim createTemporary = True
+
+                If initialValueToRewrite.Kind = Global.Microsoft.CodeAnalysis.VisualBasic.BoundKind.ObjectCreationExpression OrElse initialValueToRewrite.Kind = Global.Microsoft.CodeAnalysis.VisualBasic.BoundKind.NewT Then
```

Comment:
> Global.Microsoft.CodeAnalysis.VisualBasic. [](start = 47, length = 43)

nit: do these need to be fully qualified?

Reply:
> nit: do these need to be fully qualified?

No. Not sure why IDE does this. I'll revert this